test/extended/dr/static_pod_rollouts.go

```diff
@@ -0,0 +1,110 @@
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		g.GinkgoT().Log("debugging into the same node to add back the etcd static pod manifest")
+		err = oc.AsAdmin().Run("debug").Args("-n", etcdNamespace, "node/"+etcdTargetNodeName, "--", "chroot", "/host", "/bin/bash", "-c", "mv /var/lib/etcd-backup/etcd-pod.yaml /etc/kubernetes/manifests/").Execute()
+		o.Expect(err).ToNot(o.HaveOccurred(), "failed to add etcd static pod back to the node")
+
+		g.GinkgoT().Log("waiting for all the etcd instances to be available to return the cluster to its original state")
+		waitForReadyEtcdStaticPods(oc.AdminKubeClient(), initialEtcdPodCount)
+
+	})
+
+	g.It("is able to block the rollout of a revision when the quorum is not safe", func(ctx context.Context) {
+		var err error
+
+		g.GinkgoT().Log("debugging into the node to remove the etcd static pod manifest")
+		err = oc.AsAdmin().Run("debug").Args("-n", etcdNamespace, "node/"+etcdTargetNodeName, "--", "chroot", "/host", "/bin/bash", "-c", "mkdir -p /var/lib/etcd-backup && mv /etc/kubernetes/manifests/etcd-pod.yaml /var/lib/etcd-backup").Execute()
+		err = errors.Wrap(err, "failed to remove etcd static pod from the node")
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		g.GinkgoT().Log("waiting for the etcd pod to be removed")
+		waitForReadyEtcdStaticPods(oc.AdminKubeClient(), initialEtcdPodCount-1)
+
+		g.GinkgoT().Log("ensuring the quorum guard dependent controllers have degraded")
+		conditionTypes := []string{
+			"EtcdCertSignerControllerDegraded",
+			"EtcdEndpointsDegraded",
+			// "TargetConfigControllerDegraded",
+		}
+		o.Expect(wait.PollUntilContextTimeout(ctx, 10*time.Second, 5*time.Minute, true, func(ctx context.Context) (done bool, err error) {
+			// retrieve the operator status
+			etcdCluster, err := oc.AdminOperatorClient().OperatorV1().Etcds().Get(ctx, "cluster", metav1.GetOptions{})
+			if err != nil {
+				return true, err
+			}
+
+			for _, conditionType := range conditionTypes {
+				isConditionTrue := librarygov1helpers.IsOperatorConditionTrue(etcdCluster.Status.Conditions, conditionType)
+				if !isConditionTrue {
+					return false, nil
+				}
+			}
+			return true, nil
+		})).ToNot(o.HaveOccurred(), "expected the quorum guard dependent controllers to be degraded")
+
+		// getting the count of installer pods before triggering a rollout
+		installerPodsPreRolloutTrigger, err := e2epod.GetPods(ctx, oc.AdminKubeClient(), etcdNamespace, map[string]string{"app": "installer"})
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		g.GinkgoT().Log("setting the hardware speed to Slower to trigger a revision rollout")
+		data := fmt.Sprintf(`{"spec": {"controlPlaneHardwareSpeed": "Slower"}}`)
+		_, err = oc.AdminOperatorClient().OperatorV1().Etcds().Patch(ctx, "cluster", types.MergePatchType, []byte(data), metav1.PatchOptions{})
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		// allow some time after triggering a rollout before observing the changes in the cluster
+		time.Sleep(60 * time.Second)
```

Comment:
please also use polling here, I think we can poll a minute for any changes in the etcd/cluster CRD. If there are none an error will be returned that you can assert with `o.Expect(err).To(o.HaveOccurred()))`